### ISSUE TITLE
Postgres: Collections list page in admin UI cannot fetch data in prod but can in local

### ISSUE THREAD

**Issue by Ali-Parandeh** [payloadcms/payload#3841]
### Describe the Bug

I have three collections - media, questions and users. 

My questions collection is flat and has a blocks field that has 3 blocks only (media upload, richtext and text block for video url)

I have also turned on drafts and versioning on questions collection and use the postgres adapater.

On local everything works perfectly but as soon as I deploy to prod and visit the admin UI I see the following:
The API routes work ok however.

```bash
"error: column _questions_v__blocks_contentBlock._uuid does not exist"
```

<img width="1488" alt="Screenshot 2023-10-24 at 10 09 56" src="https://github.com/payloadcms/payload/assets/43654026/e59e2b65-9b57-4eb2-83b5-a9b525faff6b">

<img width="481" alt="Screenshot 2023-10-24 at 10 07 12" src="https://github.com/payloadcms/payload/assets/43654026/4b3f05e1-ee81-4cdd-8cd7-0f7f4a4974c7">

### To Reproduce

```typescript
import { CollectionConfig } from "payload/types";
import { Media } from "../blocks/Media";
import { Content } from "../blocks/Content";
import { Video } from "../blocks/Video";
import { slugField } from "../fields/slug";
import { populateAuthors } from "../hooks/populateAuthors";
import { populatePublishedDate } from "../hooks/populatePublishedDate";
import { docOwner } from "../fields/docOwner";


const Questions: CollectionConfig = {
  slug: "questions",
  hooks: {
    afterRead: [populateAuthors],
    beforeChange: [populatePublishedDate],
  },
  versions: {
    drafts: true,
  },
  admin: {
    useAsTitle: "question",
    defaultColumns: ["question", "icon", "difficulty", "category"],
  },
  fields: [
    {
      name: "icon",
      label: "Icon Number",
      type: "number",
    },
    {
      name: "question",
      label: "Question",
      type: "text",
      required: true,
    },
    {
      name: "difficulty",
      label: "Difficulty",
      type: "select",
      admin: {
        position: "sidebar",
      },
      options: [
        { label: "🟢 Easy", value: "Easy" },
        { label: "🟡 Medium", value: "Medium" },
        { label: "🔴 Hard", value: "Hard" },
      ],
      required: true,
    },
    {
      name: "category",
      label: "Category",
      type: "select",
      admin: {
        position: "sidebar",
      },
      options: [
        {
          label: "Option 1",
          value: "option1",
        },
      ....
      ],
      required: true,
    },
    {
      name: "publishedOn",
      admin: {
        date: {
          pickerAppearance: "dayAndTime",
        },
        position: "sidebar",
      },
      hooks: {
        beforeChange: [
          ({ siblingData, value }) => {
            if (siblingData._status === "published" && !value) {
              return new Date();
            }
            return value;
          },
        ],
      },
      type: "date",
    },
    ...docOwner(),
    slugField("question"),
    {
      name: "criteria",
      label: "Criteria",
      type: "textarea",
      required: true,
    },
    {
      name: "supportingInfo",
      label: "Supplementary Information",
      type: "blocks",
      blocks: [Content, Media, Video],
    },
  ],
};

export default Questions;

```

Content Block

```typescript
import { Block } from "payload/types";

export const Content: Block = {
  fields: [
    {
      name: "content",
      label: "Content",
      type: "richText",
    },
  ],
  slug: "contentBlock",
};

```

Media Block

```typescript
import { Block } from "payload/types";

export const Media: Block = {
  fields: [
    {
      name: "media",
      relationTo: "media",
      required: true,
      type: "upload",
    },
  ],
  slug: "mediaBlock",
};
```

Video Block

```typescript
import { Block } from "payload/types";

export const Video: Block = {
  slug: "videoBlock",
  fields: [
    {
      name: "title",
      label: "Title",
      type: "text",
      required: true,
    },
    {
      name: "url",
      label: "URL",
      type: "text",
    },
  ],
};

```

Steps to reproduce:

1. Yarn dev locally to migrate prod db
2. Deploy to prod
3. Access the admin UI on prod and locally. The local will load the collection list page but prod will show you this

<img width="1488" alt="Screenshot 2023-10-24 at 10 09 56" src="https://github.com/payloadcms/payload/assets/43654026/e59e2b65-9b57-4eb2-83b5-a9b525faff6b">

### Payload Version

^2.0.0

### Adapters and Plugins

    "@google-cloud/storage": "^7.3.0",     "@payloadcms/bundler-webpack": "^1.0.0",     "@payloadcms/db-postgres": "^0.x",     "@payloadcms/plugin-cloud": "^2.0.0",     "@payloadcms/plugin-cloud-storage": "^1.1.1",     "@payloadcms/richtext-slate": "^1.0.0",     "cross-env": "^7.0.3",     "dotenv": "^8.2.0",     "express": "^4.17.1",     "payload": "^2.0.0"

**Comment by JarrodMFlesch** [comment]
@Ali-Parandeh What version of payload shows up in your lock file?

**Comment by JarrodMFlesch** [comment]
@Ali-Parandeh Can you try running `payload migrate:create` on your prod instance?

**Comment by Ali-Parandeh** [comment]
@JarrodMFlesch Sorry I was running against deadlines so I dropped the Postgres adapter in favour of mongoose adapter and switched to mongodb instead. It is all now working as expected.

**Comment by Ali-Parandeh** [comment]
The postgres adapter is not stable enough for schema changes and can run locally but not in prod sometimes. Payload migrate create might have worked not sure

**Comment by Ali-Parandeh** [comment]
"@google-cloud/storage": "^7.3.0",     "@payloadcms/bundler-webpack": "^1.0.0",     "@payloadcms/db-postgres": "^0.x",     "@payloadcms/plugin-cloud": "^2.0.0",     "@payloadcms/plugin-cloud-storage": "^1.1.1",     "@payloadcms/richtext-slate": "^1.0.0",     "cross-env": "^7.0.3",     "dotenv": "^8.2.0",     "express": "^4.17.1",     "payload": "^2.0.0"

**Comment by JarrodMFlesch** [comment]
@Ali-Parandeh where are you deploying this at? Payload Cloud?

**Comment by Ali-Parandeh** [comment]
This is on Google cloud run on Google cloud as a docker container

**Comment by JarrodMFlesch** [comment]
@Ali-Parandeh ok, I just saw that you listed `@payloadcms/plugin-cloud` as a dependency above.

**Comment by JarrodMFlesch** [comment]
@Ali-Parandeh I was able to load this locally also, I think the migrations might have worked. I am going to close this for now since you switched from the beta db-postgres adapter to the stable db-mongodb adapter for now. If you run into this in the future feel free to open a new issue. Thanks!

**Comment by jmikrut** [comment]
Hey all, I am pretty confident that this is related to your database and production, not having the required fields as indicated by your schema. The UUID field is for version enabled collections and came with a postgres bug fix a couple months ago. Try and run migrations against prod to see if that fixes it.
